### PR TITLE
Guide for adding new platform in jetstream project

### PR DESCRIPTION
**Pull request by kik-kik**
## Description (optional)

-

## Issue that this pull request resolves (optional)
Closes: mozilla/jetstream#949

### CODE REVIEW

**Comment by kik-kik** [comment]
@scholtzan - I tried to add as much info as I could gather. I'm not familiar with the process inside experimenter, mozanalysis, and what tests are required. Would be amazing if you could share this info with me so that I can add it.

Otherwise, please let me know your thoughts.

File: docs/adding_new_platform_guide.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+---
+### Configuration breakdown
+- `[platform.platform_name]` - Specify platform name
+- `config_spec_path` - toml configuration for the app found inside `jetstream/config` folder (default: `<platform_name>.toml`).
+- `metrics_module` - mozanalysis metrics module that this platform should use (default: `<platform_name>`)
+- `segments_module` - mozanalysis segments module that this platform should use (default: `None`)
+- `enrollments_query_type` - TODO (default: `glean-event`)
+- `validation_app_id` - TODO
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:88]:
```suggestion
- `enrollments_query_type` - whether enrollments should be determined based on Glean events (`glean-event`) data or Normandy data (`normandy`) (default: `glean-event`)
- `app_id` - application ID as defined in [probe-scraper repository.yaml](https://github.com/mozilla/probe-scraper/blob/main/repositories.yaml)
```

For context: Normandy is how we rolled out experiments before using Glean.

File: docs/adding_new_platform_guide.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+pip-compile --generate-hashes --output-file=requirements.txt requirements.in
+```
+
+- Commit changes to both `requirements.in` and `requirements.txt` files
+- Push your branch to remote
+- Create a pull request to merge your branch into the main branch and wait for a review before merging
+
+---
+## Add support for platform in jetstream
+- Inside [platform_config.toml](https://github.com/mozilla/jetstream/blob/main/platform_config.toml) add specification/configuration for the new platform
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:68]:
```suggestion
- Inside [platform_config.toml](https://github.com/mozilla/jetstream/blob/main/platform_config.toml) add configuration for the new platform
```

File: docs/adding_new_platform_guide.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+- This will trigger CI pipeline which will release the new version of `mozanalysis` package to [PyPi](https://pypi.org/project/mozanalysis/)
+
+*More information about the deployment and tag format can be found [here](https://github.com/mozilla/mozanalysis#deploying-a-new-release)*
+
+---
+## Update mozanalysis in jetstream
+https://github.com/mozilla/jetstream
+
+- Make sure your main brach is up-to-date with remote
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:47]:
```suggestion
- Make sure your `main` branch is up-to-date with remote
```

File: docs/adding_new_platform_guide.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+- https://experimenter.info/jetstream/configuration#experiment-section
+
+---
+## Adding support in mozanalysis
+https://github.com/mozilla/mozanalysis/
+- Add new metric(s) in `mozanalysis` (if needed). For more detailed guide follow [how do I add a metric to my experiment](https://experimenter.info/jetstream/metrics#how-do-i-add-a-metric-to-my-experiment), and [defining metric](https://experimenter.info/jetstream/configuration#metrics-section)
+- Add new segment(s) in `mozanalysis` (if needed). For more detailed guide follow [defining segment](https://experimenter.info/jetstream/configuration#defining-segments)
+
+---
+## New mozanalysis release
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:25]:
```suggestion
```
We can probably merge these two sections

File: docs/adding_new_platform_guide.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+https://github.com/mozilla/experimenter
+
+- Follow the [run an experiment guide](https://experimenter.info/#run-an-experiment) for creating a new experiment
+- https://experimenter.info/jetstream/configuration#experiment-section
+
+---
+## Adding support in mozanalysis
+https://github.com/mozilla/mozanalysis/
+- Add new metric(s) in `mozanalysis` (if needed). For more detailed guide follow [how do I add a metric to my experiment](https://experimenter.info/jetstream/metrics#how-do-i-add-a-metric-to-my-experiment), and [defining metric](https://experimenter.info/jetstream/configuration#metrics-section)
+- Add new segment(s) in `mozanalysis` (if needed). For more detailed guide follow [defining segment](https://experimenter.info/jetstream/configuration#defining-segments)
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:22]: - Default `metric`s and `datasource`s need to be added to [mozanalysis](https://github.com/mozilla/mozanalysis/blob/main/src/mozanalysis/metrics). These default metric and data source definitions will be available for [custom experiment configurations](https://experimenter.info/jetstream/configuration/)
- Optionally, `segment` definitions can also be added in [mozanalysis](https://github.com/mozilla/mozanalysis/tree/main/src/mozanalysis/segments)

**Review by scholtzan — COMMENTED** [review]
Looking good. The main thing is to move this document into the `jetstream/` directory + some inline suggestions.

File: docs/adding_new_platform_guide.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+
+An example of desktop configuration
+```
+[platform.firefox_desktop]
+config_spec_path = "default_metrics.toml"
+metrics_module = "desktop"
+segments_module = "desktop"
+enrollments_query_type = "normandy"
+validation_app_id = "firefox-desktop"
```
[the remark this answers is not in the document]
Comment (kik-kik) [reply docs/adding_new_platform_guide.md:62]: ok cool, I can see you already update the jetstream repo. I'll include this change in this guide. Thanks 🙇

**Comment by kik-kik** [comment]
@scholtzan ok I moved the file over to the `docs/jetstream/` folder.

File: docs/adding_new_platform_guide.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+# Adding a Platform
+
+Jetstream runs analyses for experiments launched on several different platforms, such as Fenix or Firefox Desktop. When adding a new platform in Experimenter, the new platform also needs to be configured in mozanalysis and Jetstream to enable automated analyses for launched experiments.
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:3]: This file can be removed

File: docs/adding_new_platform_guide.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+<br />
+
+- Optionally, segment definitions can also be added in mozanalysis:
+    - Add new segment(s) in `mozanalysis` (if needed). For more detailed guide follow [defining segment](https://experimenter.info/jetstream/configuration#defining-segments)
+
+<br />
+
+- After adding support to mozanalysis and merging your changed into the [main branch](https://github.com/mozilla/mozanalysis/tree/main) new package needs to be published to [PyPi](https://pypi.org/project/mozanalysis/)
+- Make sure `main` branch is your current branch and create a new git tag using:
```
Comment (scholtzan) [inline docs/adding_new_platform_guide.md:20]: looks like the file in `jetstream/` doesn't have these changes

**Review by scholtzan — COMMENTED** [review]
I think you accidentally updated the old file after copying it over to the `jetstream/` directory

**Comment by kik-kik** [comment]
hmm, I'm not really sure what happened there since I moved the file locally. But indeed it's in both locations. Maybe I had it still open in my code editor and then saved it there. Anyways, thanks for spotting that @scholtzan. Now it should be fixed. 🙇

**Review by scholtzan — APPROVED** [review]
Thank you for writing this up 🚀

**Pull request merged**